include/effects/matrix/PatternWeather.h

```diff
@@ -247,41 +296,27 @@ class PatternWeather : public LEDStripEffect
         }
         auto task = pObj->weatherTask;
         pObj->weatherTask = nullptr;
+        pObj->updateInProgress = false;
         debugW("Weather thread exiting");
         vTaskDelete(task);
     }
 
+    bool HasLocationChanged()

```

Comment:
What if LAT/LONG changes, does that matter?

Reply:
LAT/LONG is only set by UpdateCoordinates(), which retrieves them from OpenWeatherAPI's geocoding endpoint using Location and Country. This function checks if the Location and Country used to retrieve the instances LAT/LONG are (still) in sync with Location and Country as set in the device settings.